packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -559,7 +571,29 @@ export class KeybindingWidget extends ReactWidget {
                 }, oldKeybinding);
             }
         });
-    }
+    };
+
+    /**
+     * Prompt users to update the `when` expression for the given command.
+     * @param item the keybinding item.
+     */
+    protected changeWhenExpression = (item: KeybindingItem) => () => {
```

Comment:
i don't understand what it going here at all. Why do we have a function which create another function?

Reply:
I've been trying to experiment and look online on how they created event-handlers which pass arguments. So far I saw that they either use the arrow notation, or the `bind` which both have issues and are against the coding-guidelines as well. I guess the only way to fix it would be a bigger refactoring to create a react component as mentioned: https://github.com/eclipse-theia/theia/pull/7848#discussion_r430200625

Reply:
react components are easy look as we use them in other places

Reply:
> react components are easy look as we use them in other places

Yes! Thank you for the feedback and guidance during the implementation.